man/systemd.exec.xml

```diff
@@ -671,12 +671,13 @@
         part of a unit for which dynamic users/groups are enabled do not leave files or directories owned by
         these users/groups around, as a different unit might get the same UID/GID assigned later on, and thus
         gain access to these files or directories. If <varname>DynamicUser=</varname> is enabled,
-        <varname>RemoveIPC=</varname> and <varname>PrivateTmp=</varname> are implied (and cannot be turned
-        off). This ensures that the lifetime of IPC objects and temporary files created by the executed
-        processes is bound to the runtime of the service, and hence the lifetime of the dynamic
-        user/group. Since <filename>/tmp/</filename> and <filename>/var/tmp/</filename> are usually the only
-        world-writable directories on a system this ensures that a unit making use of dynamic user/group
-        allocation cannot leave files around after unit termination. Furthermore
+        <varname>RemoveIPC=</varname> is implied (and cannot be turned off). This ensures that the lifetime
+        of IPC objects and temporary files created by the executed processes is bound to the runtime of the
+        service, and hence the lifetime of the dynamic user/group. Since <filename>/tmp/</filename> and
+        <filename>/var/tmp/</filename> are usually the only world-writable directories on a system, unless
+        <varname>PrivateTmp=</varname> is manually enabled, those directories will be placed on a private
```

Comment:
Hmm, if we're doing this, why not just expose the new mode? I think people might have other valid use cases, e.g. when they keep their /tmp/ on host on disk but want the unit to have a private tmpfs? It would also make this part of the documentation easier to grasp?

Reply:
I mean, we already have ProtectHome=tmpfs and stuff, so it makes sense to beef up PrivateTmp= too?

Reply:
One can always add TemporaryFileSystem=/tmp, I'd much rather not add more options as there's so many already, given this can be achieved already if needed. The purpose here is to stop the implicit behaviour with DynamicUser, which cannot be done otherwise, as it's unconditional.

Reply:
Hmm, `TemporaryFileSystem=/tmp /var/tmp` would suffer from the same multiple tmpfs problem. Also the user would need to manually set size limit and such. I'm leaning towards making this a first-class option TBH.

Reply:
That's never been an issue, one doesn't have to use either. The problem is only with the implicit behaviour, that's the purpose of this PR, I do not want to make this any more complicated that it needs to be